### ISSUE TITLE
Add missing People emojis

### ISSUE THREAD

**Issue by vanniktech** [vanniktech/Emoji#10]
Those are missing: 
💑👩‍❤️‍👩👨‍❤️‍👨💏👩‍❤️‍💋‍👩👨‍❤️‍💋‍👨👪👨‍👩‍👧👨‍👩‍👧‍👦👨‍👩‍👦‍👦👨‍👩‍👧‍👧👩‍👩‍👦👩‍👩‍👧👩‍👩‍👧‍👦👩‍👩‍👦‍👦👩‍👩‍👧‍👧👨‍👨‍👦👨‍👨‍👧👨‍👨‍👧‍👦👨‍👨‍👦‍👦👨‍👨‍👧‍👧

**Comment by rafaelekol** [comment]
How I can help in adding new emojis? I found new emojis here http://emojipedia.org/apple/ios-9.3/ . But they are bigger than the ones in your library.

**Comment by vanniktech** [comment]
If they're just 'too big' we could resize them. I got most of the emojis from [here](https://github.com/arvida/emoji-cheat-sheet.com/)

**Comment by rafaelekol** [comment]
We parsed this [Unicode site](http://unicode.org/emoji/charts/full-emoji-list.html#1f601) for Apple icons. I resized them. Here you can get them all.
[emoji_48x48.zip](https://github.com/vanniktech/Emoji/files/302572/emoji_48x48.zip)

**Comment by vanniktech** [comment]
Puh nice the only question is now how those should be mapped back into the categories?

**Comment by rafaelekol** [comment]
For naming emojis I used http://unicode.org/ naming, So I think it would be more standardised. There are minor changes.

**Comment by vanniktech** [comment]
Yes but somehow we need to put them into [categories](https://github.com/vanniktech/Emoji/tree/master/library/src/main/java/com/vanniktech/emoji/emoji).

**Comment by rafaelekol** [comment]
Oh I see, I haven't reached that part of code yet. Will share it with you if I find new solution.

**Comment by rafaelekol** [comment]
For putting new Emoji in categories, we did like this for families and flags:
```
//emoji file names
emoji_1f1e6_1f1ea.png
emoji_1f468_200d_1f469_200d_1f466_200d_1f466.png
```

```
//people
Emoji.fromUnicodeArray(new int[]{0x1F468, 0x200D, 0x1F469, 0x200D, 0x1F467}),
Emoji.fromUnicodeArray(new int[]{0x1F468, 0x200D, 0x1F469, 0x200D, 0x1F467, 0x200D, 0x1F466}),

//flags
Emoji.fromUnicodeArray(new int[]{0x1f1e6, 0x1f1eb}),
Emoji.fromUnicodeArray(new int[]{0x1f1e6, 0x1f1fd}),
```

and parsed them in Emoji class like this:
```
public static Emoji fromUnicodeArray(final int[] codePoints) {
        String unicodeString = "";
        List<String> resourceParts = new ArrayList<>();
        for (Integer codePoint : codePoints) {
            unicodeString += new String(Character.toChars(codePoint));
            resourceParts.add(Integer.toString(codePoint, 16));
        }
        return new Emoji(unicodeString, "emoji_" + TextUtils.join("_", resourceParts));
    }
```

**Comment by rafaelekol** [comment]
EmojiHandler to handle such emoji files
```
//emoji file names
emoji_1f1e6_1f1ea.png
emoji_1f468_200d_1f469_200d_1f466_200d_1f466.png
```

```
package com.example.app.ui.emoji;

import android.graphics.Paint;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;
import android.text.Spannable;
import android.text.TextUtils;
import android.text.style.DynamicDrawableSpan;

import com.example.app.App;

public final class EmojiHandler {

    private static final String TAG = "EmojiHandler";

    public static void addEmojis(final Spannable text, Paint.FontMetricsInt fontMetrics) {
        final int textLength = text.length();

        // remove spans throughout all text
        // noinspection ForLoopReplaceableByForEach
        for (EmojiSpan oldSpan : text.getSpans(0, textLength, EmojiSpan.class)) {
            text.removeSpan(oldSpan);
        }

        int skip;
        for (int i = 0; i < textLength; i += skip) {
            int tempSkip = 0;

            final int unicode = Character.codePointAt(text, i);
            skip = Character.charCount(unicode);
            String emojiName = "";

            if (unicode > 0xff) {
                String hex = Integer.toString(unicode, 16);
                emojiName = "emoji_" + hex;
            }

            //check for complex emojis(consisting of multiple unicodes)
            if (i + skip < textLength) {
                tempSkip = skip;
                final int unicodeNext = Character.codePointAt(text, i + skip);

                if ((unicode == 0x1f468 || unicode == 0x1f469) && unicodeNext == 0x200d) {
                    //check for family, couples emojis

                    int middleUnicode = unicodeNext;
                    while (middleUnicode == 0x200d && tempSkip + i < textLength) {
                        tempSkip += Character.charCount(middleUnicode);
                        int nextUnicode = Character.codePointAt(text, i + tempSkip);
                        tempSkip += Character.charCount(nextUnicode);
                        String hex2 = Integer.toString(middleUnicode, 16);
                        String hex3 = Integer.toString(nextUnicode, 16);
                        emojiName += "_" + hex2 + "_" + hex3;

                        if (nextUnicode == 0x2764) {
                            int nextUnicode2 = Character.codePointAt(text, i + tempSkip);
                            tempSkip += Character.charCount(nextUnicode2);
                            String hex4 = Integer.toString(nextUnicode2, 16);
                            emojiName += "_" + hex4;
                        }
                        if (tempSkip + i < textLength) {
                            middleUnicode = Character.codePointAt(text, i + tempSkip);
                        } else {
                            break;
                        }
                    }
                } else if (unicodeNext >= 0x1f3fb && unicodeNext <= 0x1f3ff && !TextUtils.isEmpty(emojiName)) {
                    //handle skin toned emojis
                    String hex = Integer.toString(unicodeNext, 16);
                    emojiName = emojiName + "_" + hex;
                    tempSkip += Character.charCount(unicodeNext);
                } else if (unicodeNext == 0xfe0f) {
                    //handle keycap emoji
                    tempSkip += Character.charCount(unicodeNext);
                    if (i + tempSkip < textLength) {
                        final int unicodeNext2 = Character.codePointAt(text, i + tempSkip);
                        if (unicodeNext2 == 0x20e3) {
                            String hex1 = Integer.toString(unicode, 16);
                            String hex2 = Integer.toString(unicodeNext, 16);
                            String hex3 = Integer.toString(unicodeNext2, 16);
                            emojiName = "emoji_" + hex1 + "_" + hex2 + "_" + hex3;
                            tempSkip += Character.charCount(unicodeNext2);
                        }
                    }
                } else if (unicode >= 0x1f1e6 && unicode <= 0x1f1ff){
                    //handle flags emoji
                    String hex = Integer.toString(unicodeNext, 16);
                    emojiName = emojiName + "_" + hex;
                    tempSkip += Character.charCount(unicodeNext);
                }

            }

            if (!TextUtils.isEmpty(emojiName)) {
                int emojiId = App.getContext().getResources().getIdentifier(emojiName, "drawable", App.getContext().getPackageName());
                if (emojiId != 0) {
                    skip = tempSkip > 0 ? tempSkip : skip;
                    Drawable drawable = ContextCompat.getDrawable(App.getContext(), emojiId);
                    text.setSpan(new EmojiSpan(drawable, DynamicDrawableSpan.ALIGN_BOTTOM, fontMetrics), i, i + skip, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
                }
            }
        }
    }

}

```

**Comment by vanniktech** [comment]
Looks interesting, want to submit a PR for that?

**Comment by rafaelekol** [comment]
We have changed a lot of code, so it would be difficult for me to integrate it back to your code.

**Comment by Max01010101010101** [comment]
So.. no way to integrate new emojis in the library?

**Comment by vanniktech** [comment]
Implemented by #77
